commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitTest.java

```diff
@@ -60,4 +56,25 @@ long elapsedTime() {
    int numberOfExecutions() {
       return successes.get() + failures.size();
    }
+
+   boolean exceptionsOkay(List<Throwable> throwables) {
+      return throwables.stream().allMatch(t -> {
+         do {
+            if (exceptionOkay(t)) {
+               return true;
+            }
+         } while ((t = t.getCause()) != null);
+         return false;
+      });
+   }
+
+   boolean exceptionOkay(Throwable t) {
+      if (t instanceof IllegalStateException) {
+         StackTraceElement topElement = t.getStackTrace()[0];
+         // MagicKey can cause random failures where the cluster isn't in a good state, just ignore those
+         // failures in regards to being flaky
+         return topElement.getClassName().contains("MagicKey") && topElement.getMethodName().equals("<init>");
```

Comment:
Defining an allowed flaky test this way is pretty horrible and requires everyone in the team to be aware of this class. Maybe an annotation on a class would be better?

Reply:
💯 Sure, annotation would be pretty easy to do. This was just an initial thing of describing the approach.

Reply:
Btw, not saying it is correct but our thread leaker check has a similar check that everyone has to be aware of, which is what I originally based this on.